pkg/server/plugin/nodeattestor/gcp/iit.go

```diff
@@ -0,0 +1,197 @@
+package gcp
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/hcl"
+
+	jwt "github.com/dgrijalva/jwt-go"
+	cgcp "github.com/spiffe/spire/pkg/common/plugin/gcp"
+	spi "github.com/spiffe/spire/proto/common/plugin"
+	"github.com/spiffe/spire/proto/server/nodeattestor"
+)
+
+const (
+	pluginName    = "gcp_iit"
+	audience      = "spire-gcp-node-attestor"
+	googleCertURL = "https://www.googleapis.com/oauth2/v1/certs"
+)
+
+type tokenKeyRetriever interface {
+	retrieveKey(token *jwt.Token) (interface{}, error)
+}
+
+type googlePublicKeyRetriever struct {
```

Comment:
nit: this struct and its methods are probably best defined in a dedicated file (perhaps `google_public_key_retriever.go` to match the test file?)